core/src/main/java/org/jdbi/v3/BuiltInArgumentFactory.java

```diff
@@ -125,22 +106,26 @@ public Argument build(Type expectedType, Object value, StatementContext ctx)
 
         ArgBuilder v = BUILDERS.get(expectedClass);
         if (v != null) {
-            return v.build(value);
+            return Optional.of(v.build(value));
         }
 
         // Enums must be bound as VARCHAR.
         if (value instanceof Enum) {
-            return STR_BUILDER.build((((Enum<?>)value).name()));
+            Enum<?> enumValue = (Enum<?>) value;
+            return Optional.of(STR_BUILDER.build(enumValue.name()));
         }
 
         if (value instanceof Optional) {
             Object nestedValue = ((Optional<?>)value).orElse(null);
             Type nestedType = findOptionalType(expectedType, nestedValue);
-            return ctx.argumentFor(nestedType, nestedValue);
+            return ctx.findArgumentFor(nestedType, nestedValue);
+        }
+
+        if (expectedType == Object.class) {
+            return Optional.of(OBJ_BUILDER.build(value));
         }
 
-        // Fallback to generic ObjectArgument
-        return OBJ_BUILDER.build(value);
```

Comment:
This seems to be a behavior change -- previously, if `expectedType != Object.class` and no primitive mapper is found, it would bind with `setObject`.  Now it will not make a binding.
This was the final fallback in case everything else did not match.  Is this change intentional?